src/plugins/discover/public/application/angular/discover.js

```diff
@@ -907,16 +647,11 @@ function discoverController($element, $route, $scope, $timeout, $window, Promise
     $scope.hits = resp.hits.total;
     $scope.rows = resp.hits.hits;
 
-    // if we haven't counted yet, reset the counts
-    const counts = ($scope.fieldCounts = $scope.fieldCounts || {});
-
-    $scope.rows.forEach((hit) => {
-      const fields = Object.keys($scope.indexPattern.flattenHit(hit));
-      fields.forEach((fieldName) => {
-        counts[fieldName] = (counts[fieldName] || 0) + 1;
-      });
-    });
-
```

Comment:
Logic extracted to, 
https://github.com/elastic/kibana/blob/5a1f3ef6f740315b7b41013f840a36b48cd68c66/src/plugins/discover/public/application/helpers/calc_field_counts.ts#L25
a bit of refactoring and added a test